helper/schema/resource_data.go

```diff
@@ -604,6 +607,67 @@ func (d *ResourceData) GetRawConfig() cty.Value {
 	return cty.NullVal(schemaMap(d.schema).CoreConfigSchema().ImpliedType())
 }
 
+// GetRawConfigAt is a helper method for retrieving specific values
+// from the RawConfig returned from GetRawConfig. It returns the cty.Value
+// for a given cty.Path or an error diagnostic if the value at the given path does not exist.
+//
+// GetRawConfigAt is considered advanced functionality, and
+// familiarity with the Terraform protocol is suggested when using it.
+func (d *ResourceData) GetRawConfigAt(valPath cty.Path) (cty.Value, diag.Diagnostics) {
+	rawConfig := d.GetRawConfig()
+	configVal := cty.DynamicVal
+
+	if rawConfig.IsNull() {
+		return configVal, diag.Diagnostics{
+			{
+				Severity: diag.Error,
+				Summary:  "Invalid config path",
+				Detail: "The Terraform Provider unexpectedly provided a path that does not match the current schema. " +
+					"This can happen if the path does not correctly follow the schema in structure or types. " +
```

Comment:
If `rawConfig.IsNull()`, does that mean the provided `valPath` is invalid? Seems more like the message should be geared towards "unexpected error", since raw config should always be non-null.